### PR TITLE
Adding utility to create Configmaps

### PR DESCRIPTION
**Pull request by sarveshkaushal**
This is related to #1129

### CODE REVIEW

**Comment by sarveshkaushal** [comment]
Continuation of this [PR](https://github.com/kubernetes-client/java/pull/1152).

**Comment by brendandburns** [comment]
@sarveshkaushal what are you trying to do that your workspace gets messed up. Are you trying to rebase?

Git can be sort of a challenging tool to get used to, if you let me know what you are trying to do when things get messed up, I may be able to help you with some workflows that I use.

**Comment by sarveshkaushal** [comment]
> @sarveshkaushal what are you trying to do that your workspace gets messed up. Are you trying to rebase?
> 
> Git can be sort of a challenging tool to get used to, if you let me know what you are trying to do when things get messed up, I may be able to help you with some workflows that I use.

Thanks @brendandburns  for helping me with this. Here is what happened: My fork has two branches, b1 and b2. They were cut from master branch of my fork. I made few changes in b1 and sent a PR to upstream. While review of this PR from b1 was pending a lot of new changes were merged into upstream master. 
After receiving review comments on b1, i made few changes in it. Before updating the new changes in PR, i thought of squashing the local commits. So i followed this:

1. git checkout b1
2. git add <changed files>
3. git commit -m "Few changes"
4. git rebase -i HEAD~3
5. Squashed the commits and saved the file
6. git checkout master
7. git pull upstream master
8. git checkout b1
9. git rebase master

At this stage i got so many new commits in my branch b1 and conflicts in 3-4 files. Didn't know how to resolve them.
In a nutshell i struggle to rebase an old branch from master if there are conflicts. (If automatic rebase fails)

**Comment by yue9944882** [comment]
https://github.com/kubernetes-client/java/pull/1187/files#diff-7b2dfbdea3f0b9d16171bc9dde0997fb

@sarveshkaushal how do you think about rebasing onto `kubectl create`?

**Comment by sarveshkaushal** [comment]
> https://github.com/kubernetes-client/java/pull/1187/files#diff-7b2dfbdea3f0b9d16171bc9dde0997fb
> 
> @sarveshkaushal how do you think about rebasing onto `kubectl create`? (or i can do this in a follow-up)

@yue9944882 - Let me try.

**Comment by sarveshkaushal** [comment]
> https://github.com/kubernetes-client/java/pull/1187/files#diff-7b2dfbdea3f0b9d16171bc9dde0997fb
> 
> @sarveshkaushal how do you think about rebasing onto `kubectl create`? (or i can do this in a follow-up)

I am confused on where to merge this. Any pointers?

**Comment by sarveshkaushal** [comment]
> https://github.com/kubernetes-client/java/pull/1187/files#diff-7b2dfbdea3f0b9d16171bc9dde0997fb
> 
> @sarveshkaushal how do you think about rebasing onto `kubectl create`? (or i can do this in a follow-up)

Any pointers on where to merge this?

**Comment by brendandburns** [comment]
I think it is complicated to both add this utility and augment `KubectlCreate` in a single PR. Let's merge this one and then work on extending KubectlCreate.

@yue9944882 wdyt?

File: util/src/main/java/io/kubernetes/client/Configmap.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import okhttp3.Call;
+import okhttp3.Response;
+
+/** Utility class to create Configmaps */
+public class Configmap {
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/Configmap.java:43]: Let's change the name of this to `ConfigMaps` to more closely align with things like `ByteStreams` `Collections` etc in Java.

Comment (sarveshkaushal) [reply util/src/main/java/io/kubernetes/client/Configmap.java]: I have changed the class name.

File: util/src/main/java/io/kubernetes/client/Configmaps.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+   * This method makes API call to K8 server to create configmap
+   *
+   * @param namespace
+   * @param v1ConfigMap
+   * @throws IOException
+   * @throws ApiException
+   */
+  private void createConfigmap(String namespace, V1ConfigMap v1ConfigMap)
+      throws IOException, ApiException {
+    Call call =
```
Comment (yue9944882) [inline util/src/main/java/io/kubernetes/client/Configmaps.java:171]: plz rebase and make it reuse the following functions
```
Kubectl.create()
  .apiClient(..)
  .namespace(namespace)
  .resource(...)
  .execute()
```

Comment (sarveshkaushal) [reply util/src/main/java/io/kubernetes/client/Configmaps.java:171]: @yue9944882 - Can we access classes from `extended` in `util`?. My IDE is showing error without adding dependency in Util's pom.xml

Comment (sarveshkaushal) [reply util/src/main/java/io/kubernetes/client/Configmaps.java:171]: @yue9944882 and @brendandburns - Can you please help here?

**Comment by sarveshkaushal** [comment]
Not sure why the build is failing. It's running fine in my local.
